0001-purpose-and-guidelines.md

```diff
@@ -0,0 +1,204 @@
+---
+oip: "0001"
+title: OIP Purpose and Guidelines
+status: Draft
+type: Meta
+author: Unnawut Leepaisalsuwanna <unnawut@omise.co>
+created: 2018-10-05
+---
+
+# OIP-0001: OIP Purpose and Guidelines
+
+## What is an OIP?
+
+OIP stands for OmiseGO Improvement Proposal. An OIP is a design document providing information to the OmiseGO community, or describing a new feature for all OmiseGO applications or its process or environment. The OIP should provide a concise technical specification of the feature and a rationale for the feature.
+
+The OIP author is responsible for building consensus within the community and documenting dissenting opinions.
+
+## OIP Rationale
+
+The OIPs are intended to be the primary mechanisms for proposing major new features, for collecting community input on an issue, and for documenting the design decisions that have gone into OmiseGO applications. Because the OIPs are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal.
+
+For OmiseGO implementers, OIPs are a convenient way to track the progress of their implementation. Ideally each implementation maintainer would list the OIPs that they have implemented. This will give end users a convenient way to know the current status of a given implementation or library.
+
+## OIP Types
+
+There are three types of OIP:
+
+- **A Standard Track OIP** describes any changes that affects one or more OmiseGO implementations, such as a change to the network protocol, a change in business rules, proposed application standards/conventions, or any change or addition that affects the interoperability with the OmiseGO applications. Furthermore, Standard OIPs can be broken down into the following categories.
+  - **Core** - improvements requiring a consensus fork, as well as changes that are not necessarily consensus critical but may be relevant to core dev discussions.
```

Comment:
I'm a little unclear on Core category. Wouldn't something the requires a consensus fork move into the Blockchain category? Would integration library improvements get covered in OIPs?

If we wanted to follow the spirit of the EIP categories, maybe we could do:
- **Core** - consensus improvements in the child chain, core dev discussions
- **Plasma** - improvements in the Plasma protocol. _(unless we want to include this into Core)_
- **Interface** -  includes improvements around client API/RPC specifications and standards
- **eWallet** - improvements around the eWallet application

Reply:
Good point. I've removed the "Blockchain" sub-type and added "Interface" instead.

I've also removed "Plasma" sub-type since I don't see a clear difference between Core and Plasma yet OIP-wise.